src/sentry/static/sentry/app/views/releasesV2/detail/commits/index.tsx

```diff
@@ -1,8 +1,154 @@
 import React from 'react';
+import {RouteComponentProps} from 'react-router/lib/Router';
+import styled from '@emotion/styled';
 
-type Props = {};
+import AsyncComponent from 'app/components/asyncComponent';
+import CommitRow from 'app/components/commitRow';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {Repository, Commit} from 'app/types';
+import EmptyStateWarning from 'app/components/emptyStateWarning';
+import {PanelHeader, Panel, PanelBody} from 'app/components/panels';
+import DropdownControl, {DropdownItem} from 'app/components/dropdownControl';
+import overflowEllipsisLeft from 'app/styles/overflowEllipsisLeft';
 
-// TODO(releasesV2): finish this component
-const ReleaseCommits = ({}: Props) => <div>ReleaseCommits</div>;
+import {getCommitsByRepository, CommitsByRepository} from '../utils';
+import ReleaseNoCommitData from '../releaseNoCommitData';
+
+const ALL_REPOSITORIES_LABEL = t('All Repositories');
+
+type RouteParams = {
+  orgId: string;
+  release: string;
+};
+
+type Props = RouteComponentProps<RouteParams, {}>;
+
+type State = {
+  commits: Commit[];
+  repos: Repository[];
+  activeRepo: string;
+} & AsyncComponent['state'];
+
+class ReleaseCommits extends AsyncComponent<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      activeRepo: ALL_REPOSITORIES_LABEL,
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {orgId, release} = this.props.params;
+
+    // TODO(releasesV2): we want to change this in Q2 2020 to fetch release commits filtered by project
+    // `/projects/${orgId}/myproject/releases/${encodeURIComponent(release)}/commits/`,
+    return [
+      [
+        'commits',
+        `/organizations/${orgId}/releases/${encodeURIComponent(release)}/commits/`,
```

Comment:
Do we have the ability to fetch only commits from the currently selected repository? Fetching data we might never use seems wasteful.

Reply:
I am not sure, but in any case, by default, we show commits for all repositories, so we need to fetch them anyway... Filter later just hides the rest of them.